core/src/main/scala/ch/epfl/scala/debugadapter/internal/ClassPathEntryLookUp.scala

```diff
@@ -0,0 +1,263 @@
+      val missingSourceFileClassFiles = mutable.Buffer[ClassFile]()
+
+      for (classFile <- classFiles) {
+        def recordSourceFile(sourceFile: SourceFile): Unit = {
+          classNameToSourceFile.put(classFile.fullyQualifiedName, sourceFile)
+          sourceUriToClassFiles.update(
+            sourceFile.uri,
+            sourceUriToClassFiles.getOrElse(sourceFile.uri, Seq.empty) :+ classFile
+          )
+        }
+
+        classFile.sourceName.flatMap(sourceNameToSourceFile.get).getOrElse(Seq.empty).toList match {
+          case Nil =>
+            // the source name is missing from the class file
+            // or the source file is missing from the source entr
+            missingSourceFileClassFiles.append(classFile)
+          case sourceFile:: Nil =>
+            // there is only one file with that name, it must be the right one
+            // even if its relative path does not match the class package
+            recordSourceFile(sourceFile)
+          case manySourceFiles =>
+            // there are several files with the same name
+            // we find the one whose relative path matches the class package
+            manySourceFiles.find(f => f.folderAsPackage == classFile.fullPackage) match {
+              case Some(sourceFile) => recordSourceFile(sourceFile)
+              case None =>
+                // there is no source file with the correct relative path
+                // so we try to find the right package declaration in each file
+                // it would be very unfortunate that 2 sources file with the same name
+                // declare the same package.
+                manySourceFiles.filter(s => findPackage(s, classFile.fullPackage)) match {
+                  case sourceFile :: Nil =>
+                    recordSourceFile(sourceFile)
+                  case _ =>
+                    orphanClassFiles.append(classFile)
+                }
+            }
+        }
+      }
+
+      new ClassPathEntryLookUp(
+        entry,
+        sourceUriToSourceFile,
+        sourceUriToClassFiles.toMap,
+        classNameToSourceFile.toMap,
+        missingSourceFileClassFiles,
+        orphanClassFiles
+      )
+    }
+  }
+
+  private def getAllSourceFiles(entry: SourceEntry): Seq[SourceFile] = {
+    entry match {
+      case SourceJar(jar) => 
+        withinJarFile(jar) { fileSystem =>
+          getAllSourceFiles(entry, fileSystem, fileSystem.getPath("/")).toVector
+        }
+      case SourceDirectory(directory) =>
+        getAllSourceFiles(entry, FileSystems.getDefault, directory).toSeq
+      case StandaloneSourceFile(absolutePath, relativePath) =>
+        Seq(SourceFile(entry, relativePath, absolutePath.toUri))
+    }
+  }
+
+  private def getAllSourceFiles(entry: SourceEntry, fileSystem: FileSystem, root: Path): Iterator[SourceFile] = {
+    if (Files.exists(root)) {
+      val sourceMatcher = fileSystem.getPathMatcher("glob:**.{scala,java}")
```

Comment:
```suggestion
      val sourceMatcher = fileSystem.getPathMatcher("glob:**.{scala,java,sc}")
```
do we want to future proof it if we ever implement run for Ammonite?

Reply:
That's a good idea, never thought of it before. It might require a bit of work though to add some tests. So I'll create an issue and I'll work on it after this PR is merged.

Reply:
It might be enough to just keep an issue open, so that once we want to implement run/debug it's easy to find.

Reply:
https://github.com/scalacenter/scala-debug-adapter/issues/67